frontend/src/App.vue

```diff
@@ -104,54 +107,71 @@ export default {
-        const curentPromptAnswers = _.get(this.currentPrompt, "answers")
-        const questions = _.get(this.currentPrompt, "questions", []);
+        const questions = _.get(this, "currentPrompt.questions", []);
+        const that = this;
         _.forEach(questions, question => {
           const questionName = question.name
-          if (question._default === FUNCTION) {
-            this.rpc.invoke("evaluateMethod", [[curentPromptAnswers], questionName, "default"]).then(response => {
-              question.default = response
-              if (question.answer === undefined) {
-                question.answer = question.default
-              }
-            })
-          }
-          if (question.when === FUNCTION) {
-            this.rpc.invoke("evaluateMethod", [[curentPromptAnswers], questionName, "when"]).then(response => {
-              question.isWhen = response
-            })
-          }
-          if (question._message === FUNCTION) {
-            this.rpc.invoke("evaluateMethod", [[curentPromptAnswers], questionName, "message"]).then(response => {
-              question.message = response
-            })
-          }
-          if (question._choices === FUNCTION) {
-            this.rpc.invoke("evaluateMethod", [[curentPromptAnswers], questionName, "choices"]).then(response => {
-              question.choices = response
-            })
+          
+          if (isEqual) {
+            const newAnswer = _.get(newAnswers, [questionName])
+            const oldAnswer = _.get(oldAnswers, [questionName])
+            isEqual = _.isEqual(newAnswer, oldAnswer)
           }
-          if (question.filter === FUNCTION) {
-            this.rpc.invoke("evaluateMethod", [[question.answer], questionName, "filter"]).then(response => {
-              question.answer = response
-            })
-          }
-          if (question.validate === FUNCTION) {
-            this.rpc.invoke("evaluateMethod", [[question.answer, curentPromptAnswers], questionName, "validate"]).then(response => {
-              question.isValid = (_.isString(response) ? false : response)
-              question.validationMessage = (_.isString(response) ? response : undefined)
-            })
+          
+          if (!isEqual) {
+            if (question.when === FUNCTION) {
+              that.rpc.invoke("evaluateMethod", [[newAnswers], questionName, "when"]).then(response => {
+                question.isWhen = response
+                if (question.isWhen === true) {
+                  that.updateQuestion(question, questionName, newAnswers)
+                }
+              })
+            } else if (question.isWhen === true) {
+              that.updateQuestion(question, questionName, newAnswers)
+            }
           }
         })
       }
     }
   },
   methods: {
+    updateQuestion(question, questionName, newAnswers) {
+      if (question._default === FUNCTION) {
```

Comment:
honestelly, I prefer to see all those "if"s under if (question.isWhen === true)
no need to wrap with a function here.